DSCResources/MSFT_xDhcpServerExclusionRange/MSFT_xDhcpServerExclusionRange.psm1

```diff
@@ -0,0 +1,319 @@
+
+    Validate-ResourceProperties @PSBoundParameters -Apply
+
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory)]
+        [String]$ScopeId,
+
+        [parameter(Mandatory)]
+        [String]$StartRange,
+
+        [parameter(Mandatory)]
+        [String]$EndRange,
+
+        [ValidateSet('IPv4')]
+        [String]$AddressFamily = 'IPv4',
+
+        [ValidateSet('Present','Absent')]
+        [String]$Ensure = 'Present'
+    )
+
+#region Input Validation
+
+    # Check for DhcpServer module/role
+    Assert-Module -moduleName DHCPServer
+
+    # Convert the Subnet Mask to be a valid IPAddress
+    $ScopeId = (Get-ValidIpAddress -ipString $ScopeId -AddressFamily $AddressFamily -parameterName 'ScopeId').ToString()
+
+    # Convert the Start Range to be a valid IPAddress
+    $StartRange = (Get-ValidIpAddress -ipString $StartRange -AddressFamily $AddressFamily -parameterName 'StartRange').ToString()
+
+    # Convert the End Range to be a valid IPAddress
+    $EndRange = (Get-ValidIpAddress -ipString $EndRange -AddressFamily $AddressFamily -parameterName 'EndRange').ToString()
+
+    # Check to ensure startRange is smaller than endRange
+    if($EndRange.Address -lt $StartRange.Address)
+    {
+        $errorMsg = $LocalizedData.InvalidStartAndEndRangeMessage
+        New-TerminatingError -errorId RangeNotCorrect -errorMessage $errorMsg -errorCategory InvalidArgument
+    }
+
+#endregion Input Validation
+
+    if($PSBoundParameters.ContainsKey('Debug')){ $null = $PSBoundParameters.Remove('Debug')}
+    if($PSBoundParameters.ContainsKey('AddressFamily')) {$null = $PSBoundParameters.Remove('AddressFamily')}
+
+    Validate-ResourceProperties @PSBoundParameters
+
+}
+
+function Validate-ResourceProperties
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory)]
+        [String]$ScopeId,
+
+        [parameter(Mandatory)]
+        [String]$StartRange,
+
+        [parameter(Mandatory)]
+        [String]$EndRange,
+
+        [ValidateSet('Present','Absent')]
+        [String]$Ensure = 'Present',
+
+        [Switch]$Apply
+    )
+
+    $checkExclusionRangeMessage = $LocalizedData.CheckExclusionRangeMessage
+    Write-Verbose -Message $checkExclusionRangeMessage
+
+    $dhcpExclusionRange = Get-DhcpServerv4ExclusionRange | ? {($_.ScopeId -eq $ScopeId) -and ($_.StartRange -eq $StartRange) -and ($_.EndRange -eq $EndRange)}
+
+    # Initialize the parameter collection
+    if($Apply)
+    {
+        $parameters = @{}
+    }
+
+    # dhcpExclusionRange is set
+    if($dhcpExclusionRange)
+    {
+        $TestExclusionRangeMessage = $($LocalizedData.TestExclusionRangeMessage) -f 'present', $Ensure
+        Write-Verbose -Message $TestExclusionRangeMessage
+
+        # if it should be present, test individual properties to match parameter values
+        if($Ensure -eq 'Present')
+        {
+            #region Test the ScopeId
```

Comment:
I'm not sure I follow the logic - would we ever get here? $dhcpExclusionRange will only ever _not be $null_ if the -ScopeId, StartRange _and_ EndRange are not defined. This is the sort of stuff unit tests would expose/test..